lib/posix/rwlock.c

```diff
@@ -34,6 +34,11 @@ int pthread_rwlock_init(pthread_rwlock_t *rwlock,
 	return 0;
 }
 
+int pthread_rwlockattr_setpshared(pthread_rwlockattr_t *attr, int pshared);
+{
+	(*rwlock)->status = pshared;
```

Comment:
I think what you will likely want here is something along the lines of this
```suggestion
	if (attr == NULL || !(pshared == PTHREAD_PROCESS_SHARED || pshared == PTHREAD_PROCESS_PRIVATE)) {
		return EINVAL;
	}

	rwlock->pshared = pshared;
```